File: scripts/generate-logo.py
#!/usr/bin/env python3
"""
Fitness App Logo Generator
Generates PNG logos from SVG design
Requires: pip install pillow
"""

from PIL import Image, ImageDraw
import os

# Create assets/images directory
assets_dir = os.path.join(os.path.dirname(__file__), '../assets/images')
os.makedirs(assets_dir, exist_ok=True)

def create_logo(size):
    """Create a fitness app logo with dumbbell design"""
    # Create image with gradient background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Colors
    primary_color = (69, 183, 209)  # Teal/Blue
    accent_color = (255, 107, 107)  # Red
    highlight_color = (255, 217, 61)  # Yellow
    
    # Background circle
    margin = 20
    draw.ellipse(
        [(margin, margin), (size-margin, size-margin)],
        fill=primary_color
    )
    
    # Center position
    cx, cy = size // 2, size // 2
    
    # Dumbbell left weight
    weight_size = size // 6
    draw.rectangle(
        [(cx - size//3 - weight_size//2, cy - weight_size//2),
         (cx - size//3 + weight_size//2, cy + weight_size//2)],
        fill=accent_color
    )
    
    # Dumbbell bar
    bar_height = size // 10
    draw.rectangle(
        [(cx - size//5, cy - bar_height//2),
         (cx + size//5, cy + bar_height//2)],
        fill=highlight_color
    )
    
    # Dumbbell right weight
    draw.rectangle(
        [(cx + size//3 - weight_size//2, cy - weight_size//2),
         (cx + size//3 + weight_size//2, cy + weight_size//2)],
        fill=accent_color
    )
    
    return img

def create_monochrome_logo(size):
    """Create monochrome logo for Android"""
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    cx, cy = size // 2, size // 2
    weight_size = size // 6
    bar_height = size // 10
    
    # Left weight
    draw.rectangle(
        [(cx - size//3 - weight_size//2, cy - weight_size//2),
         (cx - size//3 + weight_size//2, cy + weight_size//2)],
        fill=(0, 0, 0)
    )
    
    # Bar
    draw.rectangle(
        [(cx - size//5, cy - bar_height//2),
         (cx + size//5, cy + bar_height//2)],
        fill=(0, 0, 0)
    )
    
    # Right weight
    draw.rectangle(
        [(cx + size//3 - weight_size//2, cy - weight_size//2),
         (cx + size//3 + weight_size//2, cy + weight_size//2)],
        fill=(0, 0, 0)
    )
    
    return img

def create_background():
    """Create background for Android adaptive icon"""
    img = Image.new('RGBA', (512, 512), (230, 244, 254, 255))
    return img

try:
    print("🎨 Generating app logos...")
    
    # Generate various sizes
    sizes = [
        (192, 'icon.png'),
        (1024, 'adaptive-icon.png'),
        (192, 'android-icon-foreground.png'),
        (512, 'android-icon-monochrome.png'),
        (32, 'favicon.png'),
        (200, 'splash-icon.png'),
    ]
    
    for size, filename in sizes:
        if 'monochrome' in filename:
            logo = create_monochrome_logo(size)
        else:
            logo = create_logo(size)
        
        logo.save(os.path.join(assets_dir, filename), 'PNG')
        print(f"✓ Created {filename} ({size}x{size})")
    
    # Create background
    bg = create_background()
    bg.save(os.path.join(assets_dir, 'android-icon-background.png'), 'PNG')
    print("✓ Created android-icon-background.png")
    
    print("\n✅ All logos generated successfully!")
    print(f"📁 Location: {assets_dir}")

except ImportError:
    print("❌ Error: 'pillow' is required. Install with: pip install pillow")
except Exception as e:
    print(f"❌ Error: {str(e)}")

File: scripts/generate-logo.js
#!/usr/bin/env node

/**
 * Logo Generator Script
 * Generates app logo PNG files from SVG
 * Requires: npm install sharp
 */

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

// Create assets/images directory if it doesn't exist
const assetsDir = path.join(__dirname, '../assets/images');
if (!fs.existsSync(assetsDir)) {
  fs.mkdirSync(assetsDir, { recursive: true });
}

// SVG Logo Design (Fitness themed)
const logoSVG = `
<svg width="512" height="512" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
  <!-- Background -->
  <rect width="512" height="512" fill="#45B7D1" rx="120"/>
  
  <!-- Dumbbell shape -->
  <g transform="translate(256, 256)">
    <!-- Left weight -->
    <rect x="-180" y="-40" width="80" height="80" fill="#FF6B6B" rx="10"/>
    
    <!-- Bar -->
    <rect x="-80" y="-20" width="160" height="40" fill="#FFD93D" rx="8"/>
    
    <!-- Right weight -->
    <rect x="100" y="-40" width="80" height="80" fill="#FF6B6B" rx="10"/>
    
    <!-- Highlight -->
    <circle cx="0" cy="-80" r="30" fill="rgba(255, 255, 255, 0.3)"/>
  </g>
  
  <!-- Shine effect -->
  <circle cx="100" cy="100" r="80" fill="rgba(255, 255, 255, 0.2)"/>
</svg>
`;

// Generate PNG files
async function generateLogos() {
  try {
    console.log('🎨 Generating app logos...');

    // Icon (192x192)
    await sharp(Buffer.from(logoSVG))
      .resize(192, 192)
      .png()
      .toFile(path.join(assetsDir, 'icon.png'));
    console.log('✓ Created icon.png (192x192)');

    // Larger icon (1024x1024)
    await sharp(Buffer.from(logoSVG))
      .resize(1024, 1024)
      .png()
      .toFile(path.join(assetsDir, 'adaptive-icon.png'));
    console.log('✓ Created adaptive-icon.png (1024x1024)');

    // Android foreground (192x192)
    await sharp(Buffer.from(logoSVG))
      .resize(192, 192)
      .png()
      .toFile(path.join(assetsDir, 'android-icon-foreground.png'));
    console.log('✓ Created android-icon-foreground.png');

    // Android background
    const bgSVG = '<svg width="512" height="512" xmlns="http://www.w3.org/2000/svg"><rect width="512" height="512" fill="#E6F4FE"/></svg>';
    await sharp(Buffer.from(bgSVG))
      .resize(512, 512)
      .png()
      .toFile(path.join(assetsDir, 'android-icon-background.png'));
    console.log('✓ Created android-icon-background.png');

    // Android monochrome
    const monoSVG = `
    <svg width="512" height="512" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
      <g transform="translate(256, 256)">
        <rect x="-180" y="-40" width="80" height="80" fill="black" rx="10"/>
        <rect x="-80" y="-20" width="160" height="40" fill="black" rx="8"/>
        <rect x="100" y="-40" width="80" height="80" fill="black" rx="10"/>
      </g>
    </svg>
    `;
    await sharp(Buffer.from(monoSVG))
      .resize(512, 512)
      .png()
      .toFile(path.join(assetsDir, 'android-icon-monochrome.png'));
    console.log('✓ Created android-icon-monochrome.png');

    // Favicon (32x32)
    await sharp(Buffer.from(logoSVG))
      .resize(32, 32)
      .png()
      .toFile(path.join(assetsDir, 'favicon.png'));
    console.log('✓ Created favicon.png');

    // Splash icon (200x200)
    await sharp(Buffer.from(logoSVG))
      .resize(200, 200)
      .png()
      .toFile(path.join(assetsDir, 'splash-icon.png'));
    console.log('✓ Created splash-icon.png');

    console.log('\n✅ All logos generated successfully!');
  } catch (error) {
    console.error('❌ Error generating logos:', error.message);
  }
}

generateLogos();
